code401challenges/src/test/java/code401challenges/utilities/AnimalShelterTest.java

```diff
@@ -0,0 +1,26 @@
+package code401challenges.utilities;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+
+public class AnimalShelterTest {
+    AnimalShelter test;
+    @Before
+    public void setUp() throws Exception {
+        test = new AnimalShelter();
+        test.enqueue(new Dog("Dog 1"));
+        test.enqueue(new Dog("Dog 2"));
+        test.enqueue(new Cat("Cat 1"));
+        test.enqueue(new Cat("Cat 2"));
+;    }
+
+    @Test
+    public void dequeue() {
+        Animal dog = test.dequeue("dog");
```

Comment:
Saving your "dog" instance into an `Animal` typed variable makes it so that you no longer can access it's `name` variable because `Animal` is an interface so it has no instance variables. You can change this line to `Dog dog = test.dequeue("dog");`